src/pk_compiler.c

```diff
@@ -574,62 +574,72 @@ static void eatName(Compiler* compiler) {
   setNextToken(compiler, type);
 }
 
-// Complete lexing a number literal.
-static void eatNumber(Compiler* compiler) {
+static inline bool isValidBinNumLitChar(char c) {
+  return (c == '0') || (c == '1');
+}
 
-  // TODO: scientific literals.
+static inline bool isValidNumLitHexChar(char c) {
+  return ('0' <= c && c <= '9') || ('a' <= c && c<= 'f');
+}
 
-  // TODO: 0b1012 parsed as to number tokens : 0b101 and 2
-  // Should be invalid binary literal error (also for hex).
+static inline bool isNumLitTerminationChar(char c) {
+  return !isValidNumLitHexChar(c);
+}
```

Comment:
IMHO validating the `char` with separate functions, is unnecessary. Also, we're not reusing them (this will just increase the LOC). I'd like the way it was, as macro functions inside `eatNumber`

Reply:
fixed